### PR TITLE
Upgrade to version 3.0

### PR DESCRIPTION
**Pull request by nickjcroucher**
Happy new year! Addresses most of the improvements discussed in https://github.com/nickjcroucher/gubbins/issues/1. CI switched to Github Actions and currently passing all tests on Linux and OSX. Just restructuring of the code and packaging to address, which seems easier to deal with post-PR.

### CODE REVIEW

**Comment by degreesKelvin** [comment]
> Happy new year! Addresses most of the improvements discussed in [nickjcroucher#1](https://github.com/nickjcroucher/gubbins/issues/1). CI switched to Github Actions and currently passing all tests on Linux and OSX. Just restructuring of the code and packaging to address, which seems easier to deal with post-PR.

Hi @nickjcroucher, I have installed Gubbins in Ubuntu based on the instructions in the manual, but it is still in the 1.2.4 version and I encountered memory resource issues. Can you please provide the instructions on how I can upgrade to the latest version of Gubbins?

Thank you and Happy New Year!

Kind Regards,
Kelvin

**Comment by nickjcroucher** [comment]
Hi @degreesKelvin - I would not recommend using v3.0.0 yet, as we have not validated the accuracy of the results (coming soon hopefully!) - in the first instance, I would try installing the latest version available on conda (v2.4.1), which has some fixes to memory usage (it also uses marginal ancestral reconstruction, which may not be as accurate as the joint ancestral reconstruction in v1.2.4, but is likely to be faster and more memory efficient). If v2.4.1 doesn't work either, let me know - hopefully v3.0 will be ready for wider use soon!

**Comment by degreesKelvin** [comment]
Thank you Nick!
It is showing the correct version now:

run_gubbins.py --version
2.4.1

Kind Regards,
Kelvin 😊

File: python/gubbins/common.py [hunk 1]
```diff
@@ -185,12 +207,16 @@ def parse_and_run(input_args, program_description=""):
...
                            tree_building_command])
-            os.chdir(temp_working_dir)
-            try:
-                subprocess.check_call(tree_building_command, shell=True)
-            except subprocess.SubprocessError:
-                sys.exit("Failed while building the tree.")
-            os.chdir(current_directory)
+            if current_tree_builder == "star":
+                # Move star tree into temp dir
+                subprocess.check_call(["mv", tree_builder.tree_prefix + current_basename + tree_builder.tree_suffix, built_tree])
```
Comment (puethe) [inline python/gubbins/common.py]: Better to use library (os/shutil)

File: python/gubbins/common.py [hunk 2]
```diff
@@ -293,6 +438,149 @@ def parse_and_run(input_args, program_description=""):
...
+    elif algorithm_choice == "raxml":
+        initialised_algorithm = RAxML(threads = input_args.threads, model = model, bootstrap = input_args.bootstrap, internal_node_prefix = node_labels, verbose = input_args.verbose, additional_args = extra)
+    elif algorithm_choice == "raxmlng":
+        initialised_algorithm = RAxMLNG(threads = input_args.threads, model = model, bootstrap = input_args.bootstrap, internal_node_prefix = node_labels, verbose = input_args.verbose, additional_args = extra)
+    elif algorithm_choice == "iqtree":
+        initialised_algorithm = IQTree(threads = input_args.threads, model = model, bootstrap = input_args.bootstrap, internal_node_prefix = node_labels, verbose = input_args.verbose, additional_args = extra)
+    elif algorithm_choice == "rapidnj":
+        initialised_algorithm = RapidNJ(threads = input_args.threads, model = model, bootstrap = input_args.bootstrap, verbose = input_args.verbose, additional_args = extra)
+    elif algorithm_choice == "star":
+        initialised_algorithm = Star()
```
Comment (puethe) [inline python/gubbins/common.py:581]: no arguments passed to Star?

Comment (nickjcroucher) [reply python/gubbins/common.py:581]: It's as simple as that!

File: python/gubbins/common.py (hunk 1 above)
Comment (nickjcroucher) [reply python/gubbins/common.py]: Thanks, good point - done in cf21522.

**Comment by nickjcroucher** [comment]
Makefile removed in 2377005.

**Comment by nickjcroucher** [comment]
Further improvements suggested by @puethe:
•	Makefile.in gone in 2377005
•	Versions specified in yml in aa75c84
•	Added missing dependencies to setup.py and environment.yml in d8be9e1
•	Improved pyjar error message in 77883a8
•	Changed menu options in 51f3c46

**Pull request merged**